gcovr/__main__.py

```diff
@@ -1596,24 +1596,27 @@ def _alpha(key):
         files.append(filtered_fname)
         dirs.append(os.path.dirname(filtered_fname) + os.sep)
         cdata._filename = filtered_fname
-        ttmp = os.path.abspath(options.output).split('.')
-        longname = cdata._filename.replace(os.sep, '_')
-        longname_hash = ""
-        while True:
-            if len(ttmp) > 1:
-                cdata._sourcefile = \
-                    '.'.join(ttmp[:-1]) + \
-                    '.' + longname + longname_hash + \
-                    '.' + ttmp[-1]
-            else:
-                cdata._sourcefile = \
-                    ttmp[0] + '.' + longname + longname_hash + '.html'
-            # we add a hash at the end and attempt to shorten the
-            # filename if it exceeds common filesystem limitations
-            if len(os.path.basename(cdata._sourcefile)) < 256:
-                break
-            longname_hash = "_" + hex(zlib.crc32(longname) & 0xffffffff)[2:]
-            longname = longname[(len(cdata._sourcefile) - len(longname_hash)):]
+        if not details:
+            cdata._sourcefile = None
+        else:
+            ttmp = os.path.abspath(options.output).split('.')
```

Comment:
What specifically does this if/else fix? Can you explain, ideally by adding a comment to the code? My guess: when `options.output is None`, the detail file name calculation breaks. The result is also unneeded unless `--html-details` are requested. The `cdata._sourcefile` field is only assigned so that it can be read later when calling `html_row()` which does not use that value. Did I get this right?

Reply:
You got all that right.